modules/api/php/login.class.inc

```diff
@@ -0,0 +1,207 @@
+
+        if ($login->passwordAuthenticate($user, $password, false)) {
+            $token = $this->getEncodedToken($user);
+            if (!empty($token)) {
+                return (new \LORIS\Http\Response())
+                    ->withBody(
+                        new \LORIS\Http\StringStream(
+                            json_encode(['token' => $token])
+                        )
+                    )
+                    ->withHeader("Content-Type", "application/json")
+                    ->withStatus(200);
+            } else {
+                return (new \LORIS\Http\Response())
+                    ->withBody(
+                        new \LORIS\Http\StringStream(
+                            ['error' => 'Unacceptable JWT key']
+                        )
+                    )
+                    ->withHeader("Content-Type", "application/json")
+                    ->withStatus(500);
+            }
+        }
+        return (new \LORIS\Http\Response())
+            ->withBody(
+                new \LORIS\Http\StringStream(
+                    json_encode(array("error" => $login->_lastError))
+                )
+            )
+            ->withHeader("Content-Type", "application/json")
+            ->withStatus(401);
+    }
+
+    /**
+     * Get the SinglePointLogin class used to authenticate this request
+     * in a separate function so that it can be mocked out for testing.
+     *
+     * @return \SinglePointLogin Loris Authenticator
+     */
+    function getLoginAuthenticator()
+    {
+        return new \SinglePointLogin();
+    }
+
+    /**
+     * Return a valid JWT encoded identification token for the user
+     *
+     * @param string $user The user to return an identification token for
+     *
+     * @return string JWT encoded token
+     */
+    function getEncodedToken($user)
+    {
+        $factory = \NDB_Factory::singleton();
+        $config  = $factory->config();
+
+        $www     = $config->getSetting("www");
+        $baseURL = $www['url'];
+
+        $token = array(
+            // JWT related tokens to for the JWT library to validate
+                  "iss"  => $baseURL,
+                  "aud"  => $baseURL,
+            // Issued at
+                  "iat"  => time(),
+                  "nbf"  => time(),
+            // Expire in 1 day
+                  "exp"  => time() + 86400,
+            // Additional payload data
+                  "user" => $user,
+                 );
+
+        $key = $config->getSetting("JWTKey");
+        if (!self::isKeyStrong($key)) {
+            return "";
+        }
+        return \Firebase\JWT\JWT::encode($token, $key, "HS256");
+    }
+
+    /**
+     * Verify key meets cryptographic strength requirements
+     *
+     * @param string $key The JWT key to verify
+     *
+     * @return boolean Key passes strength test
+     */
+    static function isKeyStrong($key)
+    {
+        // Note: this code adapted from User::isPasswordStrong
+        $CharTypes = 0;
```

Comment:
You can use the occasion to correct. Not a major change

Reply:
I don't think it's a good idea to combine multiple refactorings of an API that's supposed to be stable into a single PR. It introduces too much risk of breakage. You can send another PR that changes the logic/naming (rather than the location) after this one gets merged.